store/account.go

```diff
@@ -582,8 +582,18 @@ func (m *mongoDB) UpdateProfilePOIRatingMetric(accountNumber string, poiID primi
 	if err != nil {
 		return err
 	}
-	if result.MatchedCount == 0 {
-		return errAccountNotFound
+	if result.MatchedCount == 0 { // insert the place as a non-monitored poi into user's profile
+		query = bson.M{"account_number": accountNumber}
```

Comment:
```suggestion
		query = bson.M{
			"account_number": accountNumber,	
			"points_of_interest": bson.M{
				"$not": bson.M{
					"$elemMatch": bson.M{
						"id": poiID}}},
		}
```
We need an additional condition to guarantee the `ProfilePOI` is pushed to the profile does not have it. (atomic update)